hbase-server/src/main/java/org/apache/hadoop/hbase/procedure2/store/region/RegionProcedureStore.java

```diff
@@ -304,20 +301,6 @@ private void serializeDelete(long procId, List<Mutation> mutations, List<byte[]>
     rowsToLock.add(row);
   }
 
-  /**
-   * Insert procedure may be called by master's rpc call. There are some check about the rpc call
-   * when mutate region. Here unset the current rpc call and set it back in finally block. See
-   * HBASE-23895 for more details.
-   */
-  private void runWithoutRpcCall(Runnable runnable) {
-    Optional<RpcCall> rpcCall = RpcServer.unsetCurrentCall();
-    try {
-      runnable.run();
-    } finally {
-      rpcCall.ifPresent(RpcServer::setCurrentCall);
-    }
-  }
```

Comment:
Why was this not taking care of resetting the call context?

Reply:
This is for resetting the call context. `rpcCall.ifPresent(RpcServer::setCurrentCall);`

Since it is not always called within a rpc call, the `rpcCall` is an Optional, we will only call RpcServer.setCurrentCall when the Optional contains a real RpcCall.

Reply:
Does this solved your concerns? @virajjasani

Reply:
Yes, sounds good, thanks!